app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,81 @@
+module Katello
+  class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
+    respond_to :json
+
+    before_filter :find_host_with_subscriptions
+
+    api :GET, "/hosts/:id/subscriptions", N_("List a host's subscriptions"), :deprecated => true
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def index
+      @collection = index_response
+      respond_for_index :collection => @collection
+    end
+
+    def index_response
+      entitlements = @candlepin_consumer.entitlements
+      subscriptions =  entitlements.map { |entitlement| ::Katello::SystemSubscriptionPresenter.new(entitlement) }
+      full_result_response(subscriptions)
+    end
+
+    api :PUT, "/hosts/:id/subscriptions/auto_attach", N_("Trigger an auto-attach of subscriptions")
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def auto_attach
+      sync_task(::Actions::Katello::Host::AutoAttachSubscriptions, @host)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :GET, "/hosts/:id/subscriptions/events", N_("List Candlepin events for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    def events
+      respond_for_index :collection => { :results => @candlepin_consumer.events.map { |e| OpenStruct.new(e) } }
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/remove_subscriptions"
```

Comment:
you can't do multiple subscriptions via delete, as the body doesn't seem to come across